### DISCUSSION TITLE
How to install a different version of xonsh under a different name

### DISCUSSION THREAD

**Discussion by eugenesvk** [xonsh/xonsh#5618] [Q&A]
Some googling lead to `pip install -e git+http://some_url#egg=some-name`, but the `pip install -e git+https://github.com/xonsh/xonsh@0.17.0#egg=x17` failed since it was trying to find `x17` version

Likewise, how to do the same for a local folder?

Is inconvenient venv the only way?

**Comment by anki-code** [comment]
If you're working on xonsh improvement I recommend to use IDE - https://github.com/anki-code/xonsh-developer-toolkit/?tab=readme-ov-file#setup-ide - it allows to checkout and run any version of xonsh.

Also take a look into xonsh-install - https://github.com/anki-code/xonsh-install

Finally if you have xonsh fork and you want to install xonsh from branch use:
```xsh
pip install git+https://github.com/<your account>/xonsh@<branch_name>
```

**Reply by eugenesvk** [reply]
The 2nd doesn't cover Windows, 3rd one I know, but it overwrites local install
The 1st one from the IDE guide `pip install '.[dev]'` also overwrites my locally installed version, that's what I've started with before starting to google how to install a parallel version

I'd like to have base and edited xonsh versions in parallell so I can simply run the same command in two tabs and easily compare output

**Reply by anki-code** [reply]
In Python world we have virtual environments. So when you install PyCharm CE (it's free) it will ask about creation virtual environment without affect on your system installation of python. Then you can pip install anything.

I believe that xonsh-install is also can work on Windows after some fixes because mamba supports Windows.
